packages/web3/src/config/wagmi.ts

```diff
@@ -0,0 +1,64 @@
+import { connectorsForWallets } from "@rainbow-me/rainbowkit";
+import { Config, createConfig, http } from "wagmi";
+import { celo, celoAlfajores, Chain } from "wagmi/chains";
+import { config } from "./config";
+import { cleanupStaleWalletSessions } from "./wallets";
+
+import {
+  metaMaskWallet,
+  omniWallet,
+  rabbyWallet,
+  trustWallet,
+  valoraWallet,
+  walletConnectWallet,
+} from "@rainbow-me/rainbowkit/wallets";
+import { allChains } from ".";
+
+// Avoid creating WalletConnect connectors during SSR because they rely on
+// browser-only APIs like `indexedDB`.
+const isServer = typeof window === "undefined";
+
+// Clean up stale WalletConnect sessions to prevent connection issues
+if (!isServer) {
+  cleanupStaleWalletSessions();
+}
+
+// Validate WalletConnect project ID
+if (!isServer && !config.walletConnectProjectId) {
+  console.warn(
+    "NEXT_PUBLIC_WALLET_CONNECT_ID is not set. WalletConnect functionality may not work properly.",
+  );
+}
+
+const connectors = isServer
+  ? []
+  : connectorsForWallets(
+      [
+        {
+          groupName: "Recommended for Celo chains",
+          wallets: [
+            metaMaskWallet,
+            valoraWallet,
+            walletConnectWallet,
+            omniWallet,
+            trustWallet,
+            rabbyWallet,
+          ],
+        },
+      ],
+      {
+        projectId: config.walletConnectProjectId,
+        appName: "MENTO Swap",
+      },
+    );
+
+export const wagmiConfig: Config = createConfig({
+  chains: allChains as readonly [Chain, ...Chain[]],
+
+  connectors,
+  transports: {
+    [celo.id]: http(),
+    [celoAlfajores.id]: http(),
+  },
+  ssr: true,
```

Comment:
is ssr working like this? i vaguely remember reading in the docs you'd also need this:

```suggestion
  ssr: true,
  storage: createStorage({
    storage: cookieStorage,
  })
```

but could be wrong

Reply:
Will check right now

Reply:
possibly related to this error we've been seeing very often on governance https://mento-labs.sentry.io/issues/6805194555/?alert_rule_id=16205567&alert_type=issue&notification_uuid=8c8be7aa-4309-4ea0-be26-e9569ea62c78&project=4509825739849728&referrer=discord